src/osd/PG.cc

```diff
@@ -797,11 +798,8 @@ void PG::generate_past_intervals()
     dout(10) << __func__ << " past_intervals " << past_intervals << dendl;
     // Fix it
     info.history.same_interval_since = same_interval_since;
+    dirty_info = true;
   }
-
-  // record our work.
-  dirty_info = true;
-  dirty_big_info = true;
 }
```

Comment:
This commit isn't worth it.  generate_past_intervals() is almost never called -- other PGs provide the history when they provide the info.